support/elasticsearch_search.erl

```diff
@@ -75,13 +75,17 @@ search(#search_query{search = {query, Query}, offsetlimit = Offset}, Options, Co
     ZotonicQuery = with_query_id(Query, Context),
     case is_elasticsearch(ZotonicQuery, Options) of
         true ->
-            ElasticQuery = build_query(ZotonicQuery, Offset, Context),
-            %% Optimize performance by not returning full source document
-            NoSourceElasticQuery = ElasticQuery#{<<"_source">> => false},
-            #search_result{result = Items} = SearchResult =
-                do_search(NoSourceElasticQuery, ZotonicQuery, Offset, Context),
-            Ids = [id_to_integer(Item) || Item <- Items],
-            SearchResult#search_result{result = Ids};
+	    ElasticQuery = build_query(ZotonicQuery, Offset, Context),
+	    Source = source(ZotonicQuery, false),
+	    SourceElasticQuery = ElasticQuery#{<<"_source">> => Source},
+	    #search_result{result = Items} = SearchResult =
+		do_search(SourceElasticQuery, ZotonicQuery, Offset, Context),
+	    case Source of 
+		false -> 
+		    Ids = [id_to_integer(Item) || Item <- Items],
+		    SearchResult#search_result{result = Ids};
```

Comment:
This ignores `#search_result.facets`, in which the output of the Elasticsearch response’s aggregates are placed.

Reply:
No more so than before?

Reply:
Fair enough!